src/components/pages/rules/flyouts/ruleEditor/ruleEditor.js

```diff
@@ -0,0 +1,336 @@
+      fieldOptions: [],
+      devicesAffected: 0,
+      formData
+    };
+    deviceGroupOptions = deviceGroups.map(this.toSelectOption);
+  }
+
+  componentWillUnmount() {
+    if (this.subscription) this.subscription.unsubscribe();
+  }
+
+  toSelectOption = ({ id, displayName }) => ({ label: displayName, value: id });
+
+  addCondition = () => this.conditionsLink.set([...this.conditionsLink.value, newCondition()]);
+
+  deleteCondition = (index) =>
+    (evt) => this.conditionsLink.set(this.conditionsLink.value.filter((_, idx) => index !== idx));
+
+  formIsValid() {
+    return [
+      this.ruleNameLink,
+      this.deviceGroupLink,
+      this.conditionsLink,
+    ].every(link => !link.error);
+  }
+
+  //TODO: still working on it as of 17/04/18
+  apply = (event) => {
+    event.preventDefault();
+    const { formData } = this.state;
+    console.log('TODO: Handle the form submission');
+    if (this.formIsValid()) {
+      this.setState({ isPending: true });
+      this.subscription = TelemetryService.createRule(toNewRuleRequestModel(formData))
+        .subscribe(
+          () => {
+            this.setState({ isPending: false });
+            this.props.onClose();
+          },
+          error => this.setState({ error, isPending: false })
+        );
+    }
+  }
+
+  onGroupIdChange = ({ target: { value: { value = {} } } }) => {
+    this.getDeviceCountAndFields(value);
+  }
+
+  getDeviceCountAndFields(groupId) {
+    this.props.deviceGroups.some(group => {
+      if (group.id === groupId) {
+        this.subscription = IoTHubManagerService.getDevices(group.conditions)
+          .subscribe(
+            groupDevices => {
+              this.setState({
+                fieldOptions: this.getConditionFields(groupDevices),
+                devicesAffected: groupDevices.length
+              });
+            },
+            error => this.setState({ error })
+          );
+        return true;
+      }
+      return false;
+    });
+  }
+
+  getConditionFields(devices) {
+    const conditions = new Set(); // Using a set to avoid searching the array multiple times in the every
+    devices.forEach(({ telemetry = {} }) => {
+      Object.values(telemetry).forEach(({ messageSchema: { fields } }) => {
+        Object.keys(fields).forEach((field) => {
+          if (field.indexOf('_unit') === -1) conditions.add(field);
+        })
+      })
+    })
+    return [...conditions.values()].map(field => ({ label: field, value: field }));
+  }
+
+  //todo toggle button didn't support link
+  onToggle = (event) => {
+    this.setState({ formdata: { enabled: !true } })
+  }
+
+  render() {
+    const { onClose, t } = this.props;
+    const calculationOptions = calculations.map(value => ({
+      label: t(`rules.flyouts.ruleEditor.calculation.${value}`),
+      value
+    }));
+    // Validators
+    const requiredValidator = (new Validator()).check(Validator.notEmpty, () => this.props.t('rules.flyouts.ruleEditor.validation.required'));
```

Comment:
Don't need the function as the message here since you area already in the render method.

If you keep the function and move this to the constructor that should work too.